src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,420 @@
+    "\n",
+    "For users who wish to train surrogates with IDAES tools and insert obtained models into a flowsheet, see the more versatile `SurrogateBlock()` for streamlined integration."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will begin with relevant imports. We will need basic Pyomo and IDAES components:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pytest\n",
+    "from pyomo.environ import (check_optimal_termination,\n",
+    "                           ConcreteModel,\n",
+    "                           SolverFactory,\n",
+    "                           Constraint,\n",
+    "                           Var,\n",
+    "                           Set,\n",
+    "                           value,\n",
+    "                           units as pyunits)\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.models.unit_models import SkeletonUnitModel\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from pyomo.util.check_units import assert_units_consistent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For demonstrative purposes, we will build a simple model manually defining state variables relations for two generic components. In this model, the variables specify an FcTP system where molar flow of each component, temperature and pressure are selected as state variables. Users can use this structure to write custom relations between inlet and outlet streams; for example, defining outlet pressure using an explicit correlation or outlet flow of a specific component from a known accumulation or loss in the system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "m.fs = FlowsheetBlock(default={\"dynamic\": False})\n",
+    "\n",
+    "m.fs.skeleton = SkeletonUnitModel(default={\"dynamic\": False})\n",
+    "m.fs.skeleton.comp_list = Set(initialize=[\"c1\", \"c2\"])  # two generic components\n",
+    "\n",
+    "# input vars for skeleton\n",
+    "# m.fs.time is a pre-initialized Set belonging to the FlowsheetBlock; for dynamic=False, time=[0]\n",
+    "m.fs.skeleton.flow_comp_in = Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0, units=pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.temperature_in = Var(m.fs.time, initialize=298.15, units=pyunits.K)\n",
+    "m.fs.skeleton.pressure_in = Var(m.fs.time, initialize=101, units=pyunits.kPa)\n",
+    "\n",
+    "# output vars for skeleton\n",
+    "m.fs.skeleton.flow_comp_out = Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0, units=pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.temperature_out = Var(m.fs.time, initialize=298.15, units=pyunits.K)\n",
+    "m.fs.skeleton.pressure_out = Var(m.fs.time, initialize=101, units=pyunits.kPa)\n",
+    "\n",
```

Comment:
From my experience, one of the limitations of declaring the flow, temperature and pressure variables this way is the following. In most cases, the skeleton unit model will be used in the context of a larger flowsheet. If the FcTP variables are defined in this manner, users will not be able to use `TransformationFactory` and `propagate_state`. They need to manually add the equality constraints that `TransformationFactory` adds, need to initialize the inlet state manually, and propagate the outlet state manually. So, I suggest the following. Load a thermodynamic package (any package will do), build a state block for the inlet and the outlet. 
```python
    m.fs.skeleton.properties_in = m.fs.prop_pack.build_state_block(m.fs.time, default={...})
    m.fs.skeleton.properties_out = m.fs.prop_pack.build_state_block(m.fs.time, default={...})
```
Then, add inlet and outlet ports:
```python
   # Add ports
    m.fs.skeleton.add_port(name="inlet", block=m.fs.skeleton.properties_in, doc="Inlet Port")
    m.fs.skeleton.add_port(name="outlet", block=m.fs.skeleton.properties_out, doc="Outlet Port")
```

This way, the user will be able to use `TransformationFactory` and `propogate_methods`, and the code will be clean. Also, fixing the inlet state will look familiar to the users: `m.fs.skeleton.inlet.temperature.fix(500)`

Reply:
@radhakrishnatg Could you expand on why `TransformationFactory` and `propagate_state` can not be used? As long as these are the state variables used in the rest of the flowsheet I see no reason why this wouldn't work. OF course, this assumes `FcTP` are the state variables used in the rest of the flowsheet which probably should be emphasized more (hence the need for more description of the example).

For this example, I don't think there is a need to use actual `StateBlocks`, however I do think it would be worth mentioning in the conclusion (which is currently not there) that users can include `StateBlocks` or any other IDAES components within the `SkeletonModel` as desired.

Reply:
@andrewlee94 I want to rephrase what I said. They can be used, but getting them to work is a bit tricky. 
It has been a while, so I don't remember all the details. From what I remember, the keys in the inlet and outlet ports must be the same for `TransformationFactory` and `propogate_state` to work. Even though I had the same keys, I could not get the indices on the state variables to match or something like that. At least, at that time, I struggled a lot with it. I tried many things, but nothing worked. In the end, I just gave up and started using the state blocks. That made things easier for me.

Reply:
OK, that is a problem specific issue then. Yes, the contents of the `Port` need to match exactly, but as long as that is the case then it does not matter how you construct the Vars that are contained in it.

For the purposes of this example, I think the current approach is fine, but we should emphasize that `Ports `need to match whatever they will connect with and that it is the users responsibility to make this work.

Reply:
Okay. That works. I feel that it would be useful to include an example that shows how to make ports that are compatible with the inlet/outlet ports of the remaining units. This example would have saved a lot of time and effort for me back then.